core/templates/services/contextual/url.service.ts

```diff
@@ -87,8 +87,9 @@ export class UrlService {
 
   getClassroomNameFromUrl(): string {
     let pathname = this.getPathname();
-    if (pathname.match(/\/classroom/g)) {
-      return decodeURIComponent(pathname.split('/')[2]);
+    let argumentsArray = pathname.split('/');
```

Comment:
Don't we need to validate the URL patterns?

Reply:
Well, the general form of the URL is `/<classroom_name>`, so I'm not sure if validations other than checking if the path contains only 1 `/` can be performed. The reason being that there are no hard-coded parts of the URL that can be checked for, like in the other URLs (story, topic, subtopic, etc.).